src/EventSubscriber/SendFailedMessageForRetryListener.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Answear\MessengerHeartbeatBundle\EventSubscriber;
+
+use Answear\MessengerHeartbeatBundle\Exception\RabbitMQTransportException;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerMessageFailedEvent;
+use Symfony\Component\Messenger\EventListener\SendFailedMessageForRetryListener as SymfonyForRetryListener;
+use Symfony\Component\Messenger\Exception\RejectRedeliveredMessageException;
+use Symfony\Component\Messenger\Stamp\ErrorDetailsStamp;
+
+class SendFailedMessageForRetryListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private SymfonyForRetryListener $decoratedSubscriber,
+        private ?LoggerInterface $logger = null,
+    ) {
+    }
+
+    public function onMessageFailed(WorkerMessageFailedEvent $event): void
+    {
+        $errorStamp = $event->getEnvelope()->last(ErrorDetailsStamp::class);
+
+        if (RejectRedeliveredMessageException::class === $errorStamp?->getExceptionClass()) {
+            $this->decoratedSubscriber->onMessageFailed($event);
+
+            return;
+        }
+
+        if ($this->isRabbitMQTransportException($errorStamp)) {
+            $this->logger?->info('[Keepalive] Skip message with RabbitMQTransportException from retry or critical log.');
```

Comment:
log ans skip, here on retry and on failure below.

We had something similar earlier: https://github.com/answear/messenger-heartbeat-bundle/blob/c5d58b1e13f79b6c547acafc436652ffa5df2754/src/EventSubscriber/SendFailedMessageToFailureTransportListener.php